src/overview/actions.js

```diff
@@ -31,8 +31,25 @@ export function init() {
 // Search for docs matching the current query, update the results
 export function refreshSearch({loadingIndicator = false}) {
     return async function (dispatch, getState) {
-        const { query, startDate, endDate } = ourState(getState())
-        const oldResult = ourState(getState()).searchResult
+        let {
+            query,
+            startDate: oldStartDate,
+            endDate: oldEndDate,
+            searchResult: oldResult,
+        } = ourState(getState())
+
+        const {
+            startDate: newStartDate,
+            endDate: newEndDate,
+            extractedQuery,
+        } = extractTimeFiltersFromQuery(query)
+
+        const startDate = newStartDate || oldStartDate
+        const endDate = newEndDate || oldEndDate
+        query = extractedQuery || query
+
+        if (oldStartDate !== newStartDate) dispatch(setStartDate({startDate}))
+        if (oldEndDate !== newEndDate) dispatch(setEndDate({endDate}))
```

Comment:
At the moment these two lines seems to make it impossible to cancel out NLP time queries in the overview. Like I mean if I first type `stackoverflow after "100 mins ago"`, it sets the `startDate` fine, but then if you backspace so the query is just `stackoverflow`, the `startDate` remains unchanged and continues to affect the search results. It should be cleared instead, as it's no longer present in the user's query. Let me know if that example doesn't make sense.